Content.Server/Access/Systems/IdCardSystem.cs

```diff
@@ -118,6 +118,37 @@ public bool TryChangeJobTitle(EntityUid uid, string? jobTitle, IdCardComponent?
             return true;
         }
 
+
+        public bool TryChangeJobIcon(EntityUid uid, string? jobIcon, IdCardComponent? id = null, EntityUid? player = null)
+        {
```

Comment:
Just make this take in an StatusIconPrototype and use IPrototypeManager.TryIndex<>()